lib/ipip.c

```diff
@@ -81,13 +81,17 @@ encapsulate(struct rte_mbuf *pkt, uint8_t priority,
 
 		/*
 		 * The IP header checksum filed must be set to 0
-		 * in order to offload the checksum calculation.
+		 * in order to calculate the checksum.
 		 */
 		outer_ip4hdr->hdr_checksum = 0;
-
 		pkt->l3_len = sizeof(struct rte_ipv4_hdr);
-		/* Offload checksum computation for the outer IPv4 header. */
-		pkt->ol_flags |= (PKT_TX_IPV4 | PKT_TX_IP_CKSUM);
+		pkt->ol_flags |= PKT_TX_IPV4;
+		if (likely(iface->ipv4_hw_cksum)) {
+			pkt->ol_flags |= PKT_TX_IP_CKSUM;
```

Comment:
Style: the ``if`` statement only has one line of code, we can remove the braces.